pkg/controller/elasticsearch/client/model.go

```diff
@@ -104,97 +105,34 @@ type ClusterStateNode struct {
 	} `json:"attributes"`
 }
 
-// Shards contains the shards in the Elasticsearch routing table
-// mapped to their shard number.
-type Shards struct {
-	Shards map[string][]Shard `json:"shards"`
-}
+// Shards contains the shards in the Elasticsearch cluster
+type Shards []Shard
 
-// ClusterState partially models Elasticsearch cluster state.
-type ClusterState struct {
-	ClusterName  string                      `json:"cluster_name"`
-	ClusterUUID  string                      `json:"cluster_uuid"`
-	Version      int                         `json:"version"`
-	MasterNode   string                      `json:"master_node"`
-	Nodes        map[string]ClusterStateNode `json:"nodes"`
-	RoutingTable RoutingTable                `json:"routing_table"`
+// Shard partially models Elasticsearch cluster shard.
+type Shard struct {
+	Index    string     `json:"index"`
+	Shard    string     `json:"shard"`
+	State    ShardState `json:"state"`
+	NodeName string     `json:"node"`
```

Comment:
We probably want the tag to be `nodeName` too.

Reply:
`node` is the name of the element returned by `_cat/shards`: https://www.elastic.co/guide/en/elasticsearch/reference/current/cat-shards.html

Reply:
Ah, interesting. Ok.